### ISSUE TITLE
RangeError: Maximum call stack size exceeded when called service by API Gateway is offline

### ISSUE THREAD

**Issue by PowerMogli** [feathersjs/feathers#838]
### Steps to reproduce

We are using feathers in our api gateway. Our ionic client connects to the api gateway via websocket (socket.io) and the api gateway is proxying the call with @feathersjs/client via rest to the real service 'patient'. When the service is online and reachable through our api gateway then everything works fine.
But when the service is offline we get the exception listed below.

Our proxy class
```ts
import { hooks } from '@feathersjs/authentication';
import feathers, { Application, Service } from '@feathersjs/feathers';
import rest = require('@feathersjs/rest-client');
import axios = require('axios');

import { populateAuthUser } from '../app.hooks';

export class Proxy {
  private _service: Service<any>;

  constructor(private _apiRoute: string, _upstreamUrl: string, mainApp: Application) {
    const client = feathers();
    client.configure(rest.default(_upstreamUrl).axios(axios));

    this._service = client.service(this._apiRoute);
    this._service.hooks({
      before: {
        all: [hooks.authenticate('jwt'), populateAuthUser(mainApp)],
      },
    });
  }

  public find(params) {
    return this._service.find(params);
  }

  public get(id, params) {
    return this._service.get(id, params);
  }

  public create(data, params) {
    return this._service.create(data, params);
  }

  public update(id, data, params) {
    return this._service.update(id, data, params);
  }

  public patch(id, data, params) {
    return this._service.patch(id, data, params);
  }

  public remove(id, params) {
    return this._service.remove(id, params);
  }
}
```

populateAuthUser Hook
```ts
export function populateAuthUser(mainApp: Application) {
  return async (context: HookContext<any>) => {
    const { query, user, ...rest } = context.params;

    delete user.password;
    delete user.invitedBy;
    delete user.dataUsage;

    const organization = await mainApp.service('organization').get(user.organization);

    commonHooks.paramsForServer({ query, tenantId: organization.tenantId, user });

    return context;
  };
}

```

(First please check that this issue is not already solved as [described
here](https://github.com/feathersjs/feathers/blob/master/.github/contributing.md#report-a-bug))

- [ ] Tell us what broke. The more detailed the better.
- [ ] If you can, please create a simple example that reproduces the issue and link to a gist, jsbin, repo, etc. This makes it much easier for us to debug and issues that have a reproducable example will get higher priority.

### Expected behavior
Tell us what should happen

No exception

### Actual behavior
Tell us what happens instead
Feathers application started on localhost:3000
info: after: license - Method: find
info: after: users - Method: get
info: after: authentication - Method: create
info: after: organization - Method: get
info: error: patient - Method: find: connect ECONNREFUSED 127.0.0.1:3001
error:  Error: connect ECONNREFUSED 127.0.0.1:3001
    at convert (C:\_source\work\Api\konexi.api.gateway\node_modules\@feathersjs\errors\lib\index.js:240:7)
    at toError (C:\_source\work\Api\konexi.api.gateway\node_modules\@feathersjs\rest-client\lib\base.js:6:9)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:160:7)
RangeError: Maximum call stack size exceeded
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:28:20)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
error: Unhandled Rejection at: Promise  Promise {
  <rejected> RangeError: Maximum call stack size exceeded
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:28:20)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59) } RangeError: Maximum call stack size exceeded
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:28:20)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
    at hasBinary (C:\_source\work\Api\konexi.api.gateway\node_modules\has-binary2\index.js:56:59)
### System configuration

Tell us about the applicable parts of your setup.

**Module versions** (especially the part that's not working):
Our dependencies:
"@feathersjs/authentication": "^2.1.2",
    "@feathersjs/authentication-jwt": "^2.0.0",
    "@feathersjs/authentication-local": "^1.1.0",
    "@feathersjs/configuration": "^1.0.2",
    "@feathersjs/errors": "^3.3.0",
    "@feathersjs/express": "^1.2.0",
    "@feathersjs/feathers": "^3.1.3",
    "@feathersjs/rest-client": "^1.3.3",
    "@feathersjs/socketio": "^3.2.0",
    "@feathersjs/socketio-client": "^1.1.0",
    "axios": "^0.18.0",
    "compression": "^1.7.2",
    "cors": "^2.8.4",
    "cross-env": "^5.1.3",
    "feathers-authentication-hooks": "^0.1.7",
    "feathers-hooks-common": "^4.10.0",
    "feathers-mongodb": "^3.0.1",
    "helmet": "^3.12.0",
    "lodash.clonedeep": "^4.5.0",
    "mongodb": "^3.0.4",
    "node-docker-monitor": "git://github.com/PowerMogli/node-docker-monitor.git",
    "socket.io-client": "^2.0.4",
    "winston": "^2.4.0"

**NodeJS version**: 9.8.0 or 8.10.0

**Operating System**: Windows 10

**Comment by PowerMogli** [comment]
When we switch from rest to socket.io in our proxy nothing fails. Whats the problem with rest?

**Comment by daffl** [comment]
I think the `ECONNREFUSED` error has a circular reference so the [convert method in the REST client](https://github.com/feathersjs/rest-client/blob/master/lib/base.js#L6) causes the recursive call.

**Comment by PowerMogli** [comment]
Ok. Is this an issue caused by us or is this a feathers internal issue?

**Comment by daffl** [comment]
Should be fixed now in the latest version (1.3.4) of `@feathersjs/rest-client`
